### ISSUE TITLE
View user profiles

### ISSUE THREAD

**Issue by wbazant** [falling-fruit/falling-fruit-web#504]
We would like to offer our users a way to view profiles of other users.

- add a route to the frontend, /users/:userId (we could rename the current /users/sign_in and /users/sign_up routes to /account/sign_in and /account/sign_up if it makes things easier, but user ids are numeric so there's theoretically no clash)
- display the data currently offered by the API's GET /users/{id} (https://petstore.swagger.io/?url=https://fallingfruit.org/api/0.3/openapi.yml#/Users/get_users__id_)
- link to user profile pages from location pages
- back button on the user profile page should return to a viewed location

Part of #424 - to show a list of changes made by a user, we need a dedicated user page . Also related to #496 - we'll next have a section with recent activity for a user, which can share the visual elements with recent activity for an area.

**Comment by wbazant** [comment]
Assigned to @IngvarV8 (I'll put you in the assigned field once you leave a comment here)

**Comment by IngvarV8** [comment]
@wbazant Confirming it

**Comment by wbazant** [comment]
@IngvarV8 Awesome! Github wasn't letting me assign you the issue until you commented on it ( I think they've made it like that to limit spam).

You can follow the instructions in the README.md about installing the project and running it locally and you haven't already, and if you're not sure how to get started maybe have a look at how pages like https://beta.fallingfruit.org/imports/455 are implemented, the user profile page can be similar to it except with different data.

If this issue needs clarification we can have a discussion here, or you can make other issues if something else gets in your way (e.g. if the documentation needs clarifying). And once you make some progress you can open a pull request - can be a draft - and then we can work on it together.

**Comment by IngvarV8** [comment]
@wbazant Appreciate it for the detailed info! However, the link you have provided returns some errors in Portugese (I presume this is intentional, showing only the design of the page?)

**Comment by wbazant** [comment]
I'm just showing an example imports page. We're you able to run the site locally?

**Comment by IngvarV8** [comment]
Yes, it works

**Comment by wbazant** [comment]
Awesome! do you have a plan for how to proceed with developing this feature or would you like some pointers for how to start?

**Comment by IngvarV8** [comment]
Currently I am trying to understand the whole code & technologies used. I will probably need help, but for now I'm still discovering everything :)

**Comment by wbazant** [comment]
@IngvarV8 how is it going? If you still want to solve the issue but are stuck on something, feel free to describe how far you got with the solution and any stumbling blocks, and I can try help unblock you.

**Comment by IngvarV8** [comment]
@wbazant Hello, yet I'm testing locally and I'll push it on git as soon as it's closer to being done. I skipped one week being busy and now I'm getting on it again

**Comment by IngvarV8** [comment]
@wbazant Hi, I have got a few notes/question regarding the issue:
1. locationData.user_id from inside EntryOverview.js is always null (al least I have never seen any numeric value). Thus I can't test it on real scenario yet, only by providing ids by myself directly into the code. Should I touch any API logic or will this be a part of another issue?
2. This is just for clarification, but should user page be a separate web page or just a component rendered inside entry overview? Judging by  https://beta.fallingfruit.org/imports/455 provided by you it's a separate page
3. Regardings styles, do I need to create new ones for user page or it's sufficient to use existing ones?

**Comment by wbazant** [comment]
1. Thanks for pointing that out! You're right, there's user_id for reviews and in /locations/changes, but I can see in https://petstore.swagger.io/?url=https://raw.githubusercontent.com/falling-fruit/api/main/docs/openapi.yml#/Locations/get_locations_changes that it's missing. I've made a ticket, either Ethan will add it or someone else can have a go - you can do it if you want to, or maybe I can try. For now, maybe you could just implement the users page without the correct back arrow functionality yet, and we'll test it by going to localhost:3000/users/:userId for a few values? 

 
2. You're right, we would definitely like the user page to be its own page, under the route /users/:userId. This is so we can link to it from a few places - location author, review author, and the list of recent changes (that @paskal98 is working on in #537 ), plus maybe someone will want to bookmark a link to check on a friend's account, etc.

3. We want the UI to look coherent throughout the site, but maybe there'll be a reason to do something different for this page. Maybe start with just the default styling and see what that looks like?

**Comment by IngvarV8** [comment]
@wbazant Thank you for clarification! For now I'll be using user ids directly for testing as I was (I'll switch id to locationData.user_id when I upload the code on GitHub). Regarding styles I'm sticking to existing ones then. I'll implement back arrow functionality only when  localhost:3000/users/:userId works as intended.

**Comment by ezwelty** [comment]
@IngvarV8 The API has always returned a `user_id`, but I had accidentally omitted it from the documentation, which is now fixed. To update `src/utils/apiSchema.ts` to reflect this change, run `yarn genApiSchema`.

**Comment by IngvarV8** [comment]
Okay, yesterday I uploaded changes to display user page when appending /users/:id, it's on _view-user-profiles_ branch on my fork if any of you want to test its functionality. I have modified `apiSchema.ts` (now I know I should not), but auto-generated file by `yarn genApiSchema` gives me a bunch of errors, preventing me from compiling the program. I used `apiSchema.ts` from main branch of upstream repo + a few lines of my code to make it work. I wonder if `apiSchema.ts` on GitHub main branch should be synced with auto-generated one so everyone is on the same page? Since the latter one has twice as many lines of code and does not seem compatible with the rest of the code (at least for me).
